godotproject/AntAttackFire.gd

```diff
@@ -1,24 +1,31 @@
-extends Node2D
+extends KinematicBody2D
 
-var speed = 5
-var velocity = Vector2()
+const FLOOR = Vector2(0, -1)
+const SPEED = 250
+
+var velocity = Vector2(0, 0)
 var distance_traveled = 0
 var direction = 1
+var attackDamage = 1
 
 func _physics_process(_delta):
-	velocity.x = speed * direction
+
+	velocity.x = SPEED * direction
 	velocity.y = 0 # not needed, it's already 0
-	translate(velocity)
-	distance_traveled += speed
+	velocity = move_and_slide(velocity, FLOOR)
+	distance_traveled += SPEED * _delta
 	
-	if distance_traveled > 200:
-		queue_free()
+	if distance_traveled > 200 || is_on_wall():
+		dead()
 
 func _on_PlayerCollisionArea_body_entered(body):
 	if "Player" in body.name:
-		$FireballExplodeSoundPlayer.play(0)
-		body.takeDamage(1)
-		
+		body.takeDamage(attackDamage)
+		dead()
+
+func dead():
+	$FireballExplodeSoundPlayer.play(0)
+	queue_free()
```

Comment:
This won't be able to play the sound (or rather the sound will start playing, but the fireball is also destroyed the same frame, so it will stop playing when the fireball is destroyed) @fheysel

Reply:
great insight. Ok perhaps we need to rethink this operation

Reply:
possible solution. Turn sprite invisible + disable collisions, and set Timer to destroy after the sound is completed